src/frontend/lib/uploadObject.ts

```diff
@@ -0,0 +1,53 @@
+type UploadObjectResponse = {
+    httpStatusCode: number
+    message: string
+    key?: string
+}
+
+/**
+ * Use pre-signed URL to upload file to cloud provider bucket
+ * @param presignedUrl pre-signed URL
+ * @param fields additional fields required by the pre-signed URL
+ * @param file file to upload
+ */
+export const uploadObject = async ({
+    presignedUrl,
+    fields,
+    file,
+}: {
+    presignedUrl: string
+    fields?: { [key: string]: string }
+    file: File
+}): Promise<UploadObjectResponse> => {
+    try {
+        const formData = new FormData()
+
+        // Add any additional fields required by the pre-signed URL
+        if (fields)
+            Object.entries(fields).forEach(([key, value]) => {
+                formData.append(key, value)
+            })
+        formData.append('file', file)
+
+        const response = await fetch(presignedUrl, {
+            method: 'PUT',
+            body: fields ? formData : file,
+            headers: {
+                'Content-Type': file.type,
```

Comment:
When sending FormData, manually setting the 'Content-Type' header interferes with the automatic setting of the boundary parameter, causing the request to fail.

Recommendation: Remove the 'Content-Type' header when fields are present and FormData is used. Let the browser set it automatically. Modify the code as follows:

```
const response = await fetch(presignedUrl, {
    method: 'PUT',
    body: fields ? formData : file,
    headers: fields
        ? undefined // Do not set 'Content-Type' header when sending FormData
        : {
              'Content-Type': file.type,
          },
    mode: 'cors',
    credentials: 'include',
})

```

Reply:
Thanks @BSalaeddin , done